cliff.toml

```diff
@@ -0,0 +1,99 @@
+# configuration file for git-cliff (0.1.0)
+
+[changelog]
+# changelog header
+header = """
+<!--
+Usage:
+
+Change log entries are generated by git cliff ref: https://github.com/orhun/git-cliff
+This can be run using "make changelog tag=vx.y.z"
+
+Each commit should be conventional, the following message groups are supported.
+
+* feat, feature
+* imp
+* bug, fix
+* deprecated
+* api-breaking
+* state-machine-breaking, smb
```

Comment:
these are arbitrary and can be decided by us.

Reply:
I recall some usage of a `!` prefix which would indicate an API breaking change. 
I'd like to suggest maybe adopting a `!` prefix for api breaking changes and `!!` or something else for state machine breaking. 

I feel like its common for us to use squash commits / PR titles similar to:

`refactor: some refactor of a module` or `chore: cleaning up unused to code`

But it might be nice to be able to prefix these like:

`!refactor: some refactor that breaks an api` or `!!feat: this feat introduces a state machine breaking change`.

Any thoughts on this? I'm curious to hear what others think